docs/usage/integration.md

```diff
@@ -0,0 +1,413 @@
+# Integration
+
+TigerBeetle is designed to guard against bugs not only in its [own code](../TIGER_STYLE.md), but
+at the boundaries, in the application code which interfaces with TigerBeetle.
+This is exhibited by the client's API design, which may be [surprising](#retries) when contrasted
+with a more conventional database.
+
+Strict consistency guarantees at the database level simplify application logic and error handling
+farther up the stack.
+
+
+
+## Consistency
+
+TigerBeetle provides strict serializability (serializability + linearizability) to each
+[client session](#client-sessions).
+
+But consistency models are notoriously arcane — what specifically does this mean for applications
+in practice?
+
+- [**Client sessions**](#client-sessions)
+  - A client session may have at most one in-flight request.
+  - A client session reads its own writes.
+  - A client session observes writes in the order that they occur on the cluster.
+  - A client session observes [`debits_posted`](../reference/accounts.md#debits_posted) and
+    [`credits_posted`](../reference/accounts.md#credits_posted) as monotonically increasing.
+  - A client session never observes uncommitted updates.
+  - A client session never observes a broken invariant (e.g.
+    [`flags.credits_must_not_exceed_debits`](../reference/accounts.md#flags.credits_must_not_exceed_debits)).
+  - Multiple client sessions may receive replies [out of order](#reply-order) relative to one another.
+- [**Requests**](#requests)
+  - A request executes within the cluster at most once.
+  - Requests do not [time out](#retries) — a timeout typically implies failure, which cannot be
+    conclusively determined in the context of network faults.
+  - Requests retried by their original client session receive identical replies.
+  - Requests retried by a different client (same request body, different session) may receive
+    [different replies](#consistency-with-foreign-databases).
+  - Operations within a request are executed in sequential order.
+  - Operations within a request do not interleave with other requests.
```

Comment:
This is already given as not possible because only one request can be inflight right? Unless you mean with respect to other concurrent client sessions?

Reply:
> Unless you mean with respect to other concurrent client sessions?

Yeah, that is what I meant.

How about:

>   - Operations within a request do not interleave with other requests' operations.

Reply:
Why not mention other client requests? Why keep that part implicit?

Reply:
Because I didn't think of that :sweat_smile:.

How about:

>   - Operations within a request do not interleave with operations from other clients' requests.

My only worry is that this might sound like they could interleave with operations from their own requests.

Reply:
Also, I am going to add a new bullet in this same vein:

>   - A client session can consider a request executed when it receives a reply for the request.

Reply:
> Operations within a request do not interleave with operations from other clients' requests.
> My only worry is that this might sound like they could interleave with operations from their own requests.

Yeah I'm now confused what the real behavior is.